src/scripts/main.js

```diff
@@ -1,3 +1,301 @@
 'use strict';
 
-// write your code here
+let board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+let score = 0;
+let best = 0;
+const rows = 4;
+const columns = 4;
+const start = document.getElementById('startbutton');
+
+window.onload = function() {
+  setGame();
+};
+
+start.addEventListener('click', setStart);
+
+function setStart() {
+  const elements = document.getElementsByClassName('tile');
+
+  while (elements.length > 0) {
+    elements[0].parentNode.removeChild(elements[0]);
+  }
+
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  document.getElementById('loser').classList.remove('hidden');
+  document.getElementById('winner').classList.add('hidden');
+  document.getElementById('start').classList.add('hidden');
+
+  setGame();
+}
+
+function setGame() {
+  const columnId = [];
+  const rowID = [];
+
+  for (let i = 0; i < 2; i++) {
```

Comment:
Consider using functional programming methods like `forEach` instead of `for` loops.